sockeye/arguments.py

```diff
@@ -847,7 +847,8 @@ def add_training_args(params):
                               type=int,
                               default=None,
                               help='Maximum number of samples. Default: %(default)s.')
-    train_params.add_argument(C.TRAIN_ARGS_CHECKPOINT_FREQUENCY,
+    train_params.add_argument(C.TRAIN_ARGS_CHECKPOINT_INTERVAL,
```

Comment:
Thanks for updating! how about adding the `--checkpoint-interval` argument separately and changing the existing `--checkpoint-frequency` argument to be suppressed (`help=argparse.SUPPRESS`, `default=argparse.SUPPRESS`) and be mapped to `--checkpoint-interval` directly via `dest=checkpoint_interval`? (see for example the `fill-up` argument).

This would not break anyone's training scripts but the option is now hidden and new users would simply use --checkpoint-interval. In the next major version update we could then remove the suppressed flags.